dns/rdtypes/ANY/GPOS.py

```diff
@@ -13,24 +13,36 @@
 # ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT
 # OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 
+import struct
+
 import dns.exception
 import dns.rdata
 import dns.tokenizer
+from dns._compat import long, text_type
+
 
 def _validate_float_string(what):
-    if what[0] == '-' or what[0] == '+':
+    if what[0] == b'-'[0] or what[0] == b'+'[0]:
         what = what[1:]
     if what.isdigit():
         return
-    (left, right) = what.split('.')
-    if left == '' and right == '':
+    (left, right) = what.split(b'.')
+    if left == b'' and right == b'':
         raise dns.exception.FormError
-    if not left == '' and not left.isdigit():
+    if not left == b'' and not left.decode('ascii').isdigit():
         raise dns.exception.FormError
-    if not right == '' and not right.isdigit():
+    if not right == b'' and not right.decode('ascii').isdigit():
```

Comment:
at a lot of places you just called .decode() and meant .decode('ascii') (or was it encode? well, same logic.)